File: src/main/kotlin/dev/jefy/connectpro/user/infrastructure/PasswordGenerator.java
package dev.jefy.connectpro.user.infrastructure;


import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

import dev.jefy.connectpro.user.domain.vo.EncodedPassword;
import lombok.RequiredArgsConstructor;

/**
 * @author Jôph Yamba
 */
@Service
@RequiredArgsConstructor
public class PasswordGenerator {
    private final PasswordEncoder passwordEncoder;
    
    public EncodedPassword generate(String password) {
        return new EncodedPassword(passwordEncoder.encode(password));
    }
}

File: src/main/kotlin/dev/jefy/connectpro/user/infrastructure/PasswordGenerator.kt
package dev.jefy.connectpro.user.infrastructure

import dev.jefy.connectpro.user.domain.vo.EncodedPassword
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service

/**
 * @author  Jôph Yamba
 */
@Service
class PasswordGenerator(private val passwordEncoder: PasswordEncoder) {

    fun generate(password: String): EncodedPassword {
        val encoded = passwordEncoder.encode(password)
            ?: throw CouldNotEncodePasswordException()
        return EncodedPassword(encoded)
    }
}